Tests/SwiftParserTest/AvailabilityTests.swift

```diff
@@ -178,26 +175,41 @@ final class AvailabilityTests: XCTestCase {
       func test() {}
       """,
       diagnostics: [
-        DiagnosticSpec(message: "expected version tuple in version restriction", fixIts: ["insert version tuple"]),
-        DiagnosticSpec(message: "unexpected code '0xff' in attribute"),
-      ],
-      fixedSource: """
-        @available(OSX <#integer literal#>0xff)
-        func test() {}
-        """
+        DiagnosticSpec(message: "cannot parse version 0xff")
+      ]
     )
 
     assertParse(
       """
       @available(OSX 1.0.1️⃣0xff)
       func test() {}
       """,
+      diagnostics: [
+        DiagnosticSpec(message: "cannot parse version 0xff")
+      ]
+    )
+
+    assertParse(
+      """
+      @available(OSX 1.0.1️⃣0xff, *)
+      func test() {}
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "cannot parse version 0xff")
+      ]
+    )
+
+    assertParse(
+      """
+      @available(OSX 1.0.1️⃣asdf)
+      func test() {}
+      """,
       diagnostics: [
         DiagnosticSpec(message: "expected integer literal in version tuple", fixIts: ["insert integer literal"]),
-        DiagnosticSpec(message: "unexpected code '0xff' in attribute"),
+        DiagnosticSpec(message: "unexpected code 'asdf' in attribute"),
       ],
       fixedSource: """
-        @available(OSX 1.0.<#integer literal#>0xff)
+        @available(OSX 1.0.<#integer literal#>asdf)
         func test() {}
         """
     )
```

Comment:
I have a question I would like to ask you. 
When checking for unexpected tokens, if only integers and floating literals that are included in the VersionTupleWhiteList are considered, all existing test cases generate the intended diagnosis. 
However, when there is a string literal like `asdf`, it generates a different diagnosis. 
Should we consider this as a separate case? 🤔

Reply:
If you want, you could also eat identifiers in addition to integer and float literals when consuming the unexpected version components.

Reply:
Oh, thank you! It's working just as I expected.